src/lib.rs

```diff
@@ -58,6 +58,16 @@ impl TreeState {
         self.opened.remove(identifier)
     }
 
+    /// Toggles a tree node.
+    /// If the node is in opened then it calls `close()`. Otherwise it calls `open()`.
+    pub fn toggle(&mut self) -> bool {
+        if self.opened.contains(&self.selected()) {
+            self.close(&self.selected())
+        } else {
+            self.open(self.selected())
```

Comment:
Open and close return if they changed something. This is always true for toggle so no need to return it.
```suggestion
    pub fn toggle(&mut self) {
        if self.opened.contains(&self.selected()) {
            self.close(&self.selected());
        } else {
            self.open(self.selected());
```